### ISSUE TITLE
Return value of run() and its documentation

### ISSUE THREAD

**Issue by jywarren** [publiclab/image-sequencer#41]
in the docs, we see how to use `run()` but not what it returns -- i'm assuming it's the final image state, right? Just good to be explicit, thanks!

**Comment by ccpandhare** [comment]
Umm... run() doesn't return anything... Infact most methods don't return
anything as of now.
run() just calls the draw() method on all steps

**Comment by jywarren** [comment]
maybe run should? If it's asynchronous, then you could pass it an
onComplete method. What do you think?

**Comment by ccpandhare** [comment]
`run()` already accepts a callback function. I think an `onComplete` type syntax could actually work wonders.
```
sequencer.run().onComplete(funnction(out){
    //do something with "out"
});
```

**Comment by jywarren** [comment]
ah, but wouldn't there be an order issue with run possibly completing
before onComplete was set?  I didn't see the callback documented at
https://github.com/publiclab/image-sequencer#running-the-sequencer, so
maybe I'm just behind the times.

Anyhow, great!

**Comment by ccpandhare** [comment]
Thanks a lot for pointing that out!
I don't know how I missed that there.
Looks like I need to make quite a few changes to the `Readme`. Will open a PR for that tomorrow.

As for the return issue, I will look into the applications, benefits and shortcomings of each of the methods discussed here and will come up with a solution by tomorrow.

**Comment by jywarren** [comment]
Super. Thank you!

**Comment by ccpandhare** [comment]
I think it would be better to have a syntax like
```js
sequencer.run(function(out){
    //do something with "out"
});
```
Where `out` is the output of final module.
There won't be any issue of order in this way.

Giving run a return value may not work, as run is more of an asynchronous function.
Also, it will render `run` unchainable thereafter (run is usually the last method, but still)

**Comment by jywarren** [comment]
yes re: unchainable, but that's OK!

**Comment by ccpandhare** [comment]
No, I'd like to clear myself here. 
This will work and allow chaining, and this is what we should do:
```js
sequencer.run(function(out){
    // do something
});
```

This will neither work nor allow chaining:
```js
function run() {
...
return this; // this refers to the sequencer here
}
```

So I will start working on the first one right away.

**Comment by jywarren** [comment]
oh huh, great!

**Comment by ccpandhare** [comment]
Fixed in #47
